driver/src/contracts/mod.rs

```diff
@@ -26,3 +29,36 @@ fn method_defaults(key: PrivateKey, network_id: u64) -> Result<MethodDefaults> {
 
     Ok(defaults)
 }
+
+pub enum BenignSolutionFailure {
+    BetterSolutionAlreadySubmitted,
+    NegativeUtility,
+}
+
+impl fmt::Display for BenignSolutionFailure {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            BenignSolutionFailure::BetterSolutionAlreadySubmitted => {
+                write!(f, "Better solution already submitted")
+            }
+            BenignSolutionFailure::NegativeUtility => write!(f, "Negative Utility"),
+        }
+    }
+}
+
+impl TryFrom<&str> for BenignSolutionFailure {
+    type Error = anyhow::Error;
+
+    fn try_from(reason: &str) -> Result<Self, Self::Error> {
+        match reason {
+            "New objective doesn\'t sufficiently improve current solution" => {
```

Comment:
I would be happy if we could take this opportunity to fix the \ in the logs... perhaps a double slash would correctly escape this? or do we leave it this way so it is "searchable" in plain text?

Reply:
The backslash comes from the way the string is escaped in RPC I believe. I actually think it gets printed correctly on the console:

> Feb 25 16:39:29.975 INFO [driver::driver::stablex_driver] Benign failure while verifying solution: Claimed objective doesn't sufficiently improve current solution